java-shared-dependencies/dependency-analyzer/src/main/java/com/google/cloud/DependencyAnalyzer.java

```diff
@@ -0,0 +1,119 @@
+    while (!queue.isEmpty()) {
+      VersionKey versionKey = queue.poll();
+      dependencies.add(versionKey);
+      List<VersionKey> directDependencies = depsDevClient.getDirectDependencies(versionKey);
+      // only add unseen dependencies to the queue.
+      directDependencies
+          .stream()
+          .filter(seenPackage::add)
+          .forEach(queue::offer);
+    }
+
+    List<PackageInfo> result = new ArrayList<>();
+    for (VersionKey versionKey : dependencies) {
+      QueryResult packageInfo = depsDevClient.getQueryResult(versionKey);
+      List<String> licenses = new ArrayList<>();
+      List<Advisory> advisories = new ArrayList<>();
+      for (Result res : packageInfo.getResults()) {
+        Version version = res.getVersion();
+        licenses.addAll(version.getLicenses());
+        for (AdvisoryKey advisoryKey : version.getAdvisoryKeys()) {
+          advisories.add(depsDevClient.getAdvisory(advisoryKey.getId()));
+        }
+      }
+
+      result.add(new PackageInfo(versionKey, licenses, advisories));
+    }
+
+    return new AnalyzeReport(root, result);
+  }
+
+  /**
+   * For a given package, checks the package information via <a
+   * href="https://deps.dev/">deps.dev</a> and reports error if risk information is found.
+   * <p>
+   * The types of risk checked by the dependency analyzer are:
+   * <p>1. Non-compliant licenses</p>
+   * <p>2. Security vulnerability</p>
+   * The analyzer will report all types of risk before existing with error.
+   *
+   * @param args a package. A string array with three elements.
+   * <p>The 1st element is the package management system, e.g., maven, npm, etc.</p>
+   * <p>The 2nd element is the package name.</p>
+   * <p>The 3rd element is the package version.</p>
+   * @throws IllegalArgumentException if the format of package name is incorrect according to the
+   * package management system.
+   */
+  public static void main(String[] args) throws IllegalArgumentException {
+    checkArgument(args.length == 3,
+        "The length of the inputs should be 3.\n" +
+            "The 1st input should be the package management system.\n" +
+            "The 2nd input should be the package name.\n" +
+            "The 3rd input should be the package version.\n"
+    );
+
+    String system = args[0];
+    String packageName = args[1];
+    String packageVersion = args[2];
+    DependencyAnalyzer dependencyAnalyzer = new DependencyAnalyzer(
+        new DepsDevClient(HttpClient.newHttpClient()));
+    AnalyzeReport analyzeReport = null;
+    try {
+      analyzeReport = dependencyAnalyzer.analyze(system, packageName, packageVersion);
+    } catch (URISyntaxException | IOException | InterruptedException ex) {
+      System.out.println(
+          "Caught exception when fetching package information from https://deps.dev/");
+      ex.printStackTrace();
+      System.exit(1);
+    }
+
+    AnalyzeResult result = analyzeReport.generateReport();
+    System.out.println(result);
```

Comment:
It's strange to print enum here (SUCCESS or FAIL). I want to see the report of which artifacts failed.

Reply:
The report is shown when running `generateReport` method. The result is used to determine the exit code.